amaranth/sim/_async.py

```diff
@@ -72,9 +73,18 @@ def __init__(self, interval):
         # Note: even though it is likely to be a bad idea, ``await ctx.delay(0)`` is accepted.
         # This is because, if disallowed, people are likely to do even worse things, such as
         # `await ctx.delay(1e-15)` instead.
-        if interval < 0:
+
+        if not isinstance(interval, Period):
+            # TODO(amaranth-0.7): remove
+            warnings.warn(
+                f"Per RFC 66, the `interval` argument of `DelayTrigger()` will only accept a `Period`"
+                f" in the future.",
+                DeprecationWarning, stacklevel=1)
+            interval = Period(s = interval)
+
+        if interval < Period():
             raise ValueError(f"Delay cannot be negative")
-        self.interval_fs = round(float(interval) * 1e15)
+        self.interval_fs = interval.femtoseconds
```

Comment:
it might be better to just store the `Period` directly, now that we have that type